CDP4Common/AutoGenDto/ArchitectureDiagram.cs

```diff
@@ -0,0 +1,180 @@
+        /// </param>
+        /// <param name="rev">
+        /// The revision number.
+        /// </param>
+        public ArchitectureDiagram(Guid iid, int rev) : base(iid: iid, rev: rev)
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the unique identifier of the referenced Owner.
+        /// </summary>
+        [UmlInformation(aggregation: AggregationKind.None, isDerived: false, isOrdered: false, isNullable: false, isPersistent: true)]
+        [DataMember]
+        public Guid Owner { get; set; }
+
+        /// <summary>
+        /// Gets or sets the unique identifier of the referenced TopArchitectureElement.
+        /// </summary>
+        [UmlInformation(aggregation: AggregationKind.None, isDerived: false, isOrdered: false, isNullable: true, isPersistent: true)]
+        [DataMember]
+        public Guid? TopArchitectureElement { get; set; }
+
+        /// <summary>
+        /// Gets the route for the current <see ref="ArchitectureDiagram"/>.
+        /// </summary>
+        public override string Route
+        {
+            get { return this.ComputedRoute(); }
+        }
+
+        /// <summary>
+        /// Instantiate a <see cref="CDP4Common.DiagramData.ArchitectureDiagram"/> from a <see cref="ArchitectureDiagram"/>
+        /// </summary>
+        /// <param name="cache">The cache that stores all the <see cref="CommonData.Thing"/>s</param>.
+        /// <param name="uri">The <see cref="Uri"/> of the <see cref="CDP4Common.DiagramData.ArchitectureDiagram"/></param>.
+        /// <returns>A new <see cref="CommonData.Thing"/></returns>
+        public override CommonData.Thing InstantiatePoco(ConcurrentDictionary<CacheKey, Lazy<CommonData.Thing>> cache, Uri uri)
+        {
+            return new CDP4Common.DiagramData.ArchitectureDiagram(this.Iid, cache, uri);
+        }
+
+        /// <summary>
+        /// Resolves the properties of a copied <see cref="Thing"/> based on the original and a collection of copied <see cref="Thing"/>.
+        /// </summary>
+        /// <param name="originalThing">The original <see cref="Thing"/></param>.
+        /// <param name="originalCopyMap">The map containig all instance of copied <see cref="Thing"/>s with their original</param>.
+        public override void ResolveCopy(Thing originalThing, IReadOnlyDictionary<Thing, Thing> originalCopyMap)
+        {
+            var original = originalThing as ArchitectureDiagram;
+            if (original == null)
+            {
+                throw new InvalidOperationException("The originalThing cannot be null or is of the incorrect type");
+            }
+
+            foreach (var guid in original.Bounds)
+            {
+                var copy = originalCopyMap.SingleOrDefault(kvp => kvp.Key.Iid == guid);
+                if (Equals(copy, default(KeyValuePair<Thing, Thing>)))
+                {
+                    throw new InvalidOperationException(string.Format("The copy could not be found for {0}", guid));
```

Comment:
string interpolation based on $"" has my preference of string.Format

Reply:
I've added a new issue for this, that we can do outside of this PR: https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/issues/226